pkg/userdata/scripts/health-monitor.sh

```diff
@@ -77,10 +77,18 @@ function kubelet_monitoring {
   sleep 120
   local -r max_seconds=10
   local output=""
+  local failed=false
   while [ 1 ]; do
-    if ! output=$(curl -m "${max_seconds}" -f -s -S http://127.0.0.1:10248/healthz 2>&1); then
+    if journalctl -u kubelet -n 1 | grep -q "use of closed network connection"; then
+      failed=true
+      echo "Kubelet stopped posting node status. Restarting"
+    elif ! output=$(curl -m "${max_seconds}" -f -s -S http://127.0.0.1:10248/healthz 2>&1); then
+      failed=true
       # Print the response and/or errors.
-      echo $output
+      echo "$output"
+    fi
+
+    if [ $failed ]; then
```

Comment:
there are no booleans in shell, the `if` body will always evaluate unconditionally.